engine/source/public/core/fileio.hpp

```diff
@@ -0,0 +1,47 @@
+﻿#pragma once
+#include <map>
+#include <string_view>
+#include <vector>
+
+namespace lumina
+{
+    class Engine;
+
+    class FileIO
+    {
+    public:
+        FileIO();
+        ~FileIO() = default;
+
+        FileIO(const FileIO&)            = delete;
+        FileIO(FileIO&&)                 = delete;
+        FileIO& operator=(const FileIO&) = delete;
+        FileIO& operator=(FileIO&&)      = delete;
+
+        enum class Directory
+        {
+            Assets,
+            Config,
+            Log
+        };
```

Comment:
Do we want to move the enum to the top of the `public:`? before the ctor and dtor